### PR TITLE
DOC : PyArray_Descr.names undocumented

### PR DESCRIPTION
**Pull request by vrindaaa**
Issue #10585
Updating the definition of PyArray_Descr according to the [full definition](https://github.com/numpy/numpy/blob/master/numpy/core/include/numpy/ndarraytypes.h#L591-L651)

### CODE REVIEW

File: doc/source/reference/c-api.types-and-structures.rst [hunk 1]
```diff
@@ -377,13 +380,28 @@ PyArrayDescr_Type
     normally a Python string. These tuples are placed in this
     dictionary keyed by name (and also title if given).
 
+.. c:member:: PyObject *PyArray_Descr.names
+
+    A table of functions specifiv for each
```
Comment (ahaldane) [inline doc/source/reference/c-api.types-and-structures.rst]: This isn't the right description. I think it should be the one from the comment *above* (not below) the field, the one about "An ordered tuple of field names or NULL if no fields are defined"

File: doc/source/reference/c-api.types-and-structures.rst [hunk 2]
```diff
@@ -377,13 +380,28 @@ PyArrayDescr_Type
...
 
+.. c:member:: PyObject *PyArray_Descr.metadata
+
+    Metadata specific to the C implementation
+    of the particular dtype. Added for NumPy 1.7.0.
+
+.. c:member:: NpyAuxData *PyArray_Descr.c_metadata
+
+    A pointer cached with the hash value. It is -1 if not yet
+    computed. Added for NumPy 2.0.0.
```
Comment (ahaldane) [inline doc/source/reference/c-api.types-and-structures.rst]: Hmm, I'm not sure what this field actually is. First of all, NumPy 2.0.0 doesn't exist. Second of all, as far as I can see this field is only ever set to the value -1, in other words it is never really used. That happens in `numpy/core/src/multiarray/hashdescr.c` and `numpy/core/src/multiarray/descriptor.c`.

Unless someone knows that this is actually used somewhere, probably we should just say "Currently unused. Reserved for future use in cashing hash values.".

Comment (ahaldane) [reply doc/source/reference/c-api.types-and-structures.rst]: Also, that description should be for `hash`, not for `c_metadata`.  I think you need to shift the definitions for `metadata` and `c_metadata` down by one field, and add the `hash` field.

File: doc/source/reference/c-api.types-and-structures.rst [hunk 3]
```diff
@@ -377,13 +380,32 @@ PyArrayDescr_Type
     normally a Python string. These tuples are placed in this
     dictionary keyed by name (and also title if given).
 
+.. c:member:: PyObject *PyArray_Descr.names
+
+    An ordered tupple of field names. It is NULL if no field is
```
Comment (ahaldane) [inline doc/source/reference/c-api.types-and-structures.rst]: typo in tuple

**Comment by ahaldane** [comment]
After the typo, looks good!

**Comment by ahaldane** [comment]
There appears to be some kind of communication issue with Travis CI. The test is complete but github doesn't register it.

I'll wait a little longer to see if it clears itself up.

**Comment by vrindaaa** [comment]
@ahaldane alright!

**Comment by ahaldane** [comment]
Closed and reopened to reset the tests.

But now azure-pipeline is failing with a spurious error: "The job running on agent Hosted Agent has exceeded the maximum execution time of 60."

I'm going to close and reopen one more time.

**Comment by vrindaaa** [comment]
@ahaldane any idea why these checks are failing?

**Comment by mattip** [comment]
it seems to be timing out when downloading packages to update the image for 32-bit linux on azure. I don't remember seeing this on other PRs, but it should not be specific to yours.

**Comment by ahaldane** [comment]
Seems to be an issue with ubuntu's servers currently, I can see many other people on twitter and ubuntu forums mentioning it.

Hopefully the ubuntu maintainers will have it fixed soon.

**Comment by tylerjereddy** [comment]
Probably one of the drawbacks of using our own selected Docker container on the Azure infrastructure; has behaved reasonably well up until now though; I suppose we could consider hosting a backup image on Dockerhub or something if it becomes too problematic.

**Comment by charris** [comment]
close/reopen.

**Comment by mattip** [comment]
32 bit builds are still failing. It seems not to be all that widespread, I could not find many hits on duckduckgo.

@ahaldane did you mean this [forum post](https://ubuntuforums.org/showthread.php?t=2415901)?

**Comment by ahaldane** [comment]
I had found a 2 other ubuntu forum posts, and also found these:
https://twitter.com/search?l=&q=ubuntu.com%20down%20since%3A2019-04-01&src=typd
https://www.reddit.com/r/Ubuntu/comments/b8jw0s/archiveubuntucom_barely_functioning/
https://ubuntuforums.org/showthread.php?t=2415899

There don't seem to be people on twitter mentioning it today Apr 3...

**Comment by mattip** [comment]
The build succeeded in the end, thanks @vrindaaa

**Pull request merged**